xapian-core/tests/api_weight.cc

```diff
@@ -669,6 +669,35 @@ DEFINE_TESTCASE(tfidfweight3, backend) {
     return true;
 }
 
+// Test to check default normalization is used when no values provided for parameters s and delta.
+DEFINE_TESTCASE(tfidfweight4, !backend) {
+    /* If no values provided for parameters then default normalization ("ntn") should be used */
+    Xapian::TfIdfWeight wt2;
+    TEST_EQUAL(wt2.serialise(), Xapian::TfIdfWeight("ntn").serialise());
+
+    return true;
+}
+
+// Feature tests for pivoted normalization functions.
+DEFINE_TESTCASE(tfidfweight5, backend) {
+    Xapian::Database db = get_database("apitest_simpledata");
+    Xapian::Enquire enquire(db);
+    Xapian::Query query("paragraph");
+    Xapian::MSet mset;
+
+    // Check for "PPP" normalization string.
+    enquire.set_query(query);
+    enquire.set_weighting_scheme(Xapian::TfIdfWeight("PPP"));
+    Xapian::TfIdfWeight(0.2, 1.0);
```

Comment:
No, it creates one `TfIdfWeight` using `"PPP"` as the normalisations, then another with `0.2, 1.0` as the two params (and then throws the second one away because you don't assign it to anything).

If you break line 690 down, you can assign the `TfIdfWeight` to a variable so you can play with it directly — however you can't set the parameters using the current API, so this isn't helpful. But I do think that a single constructor is the right approach.